agent/envoyextensions/builtin/otel-access-logging/structs.go

```diff
@@ -0,0 +1,434 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package otelaccesslogging
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	envoy_cluster_v3 "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	envoy_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	envoy_endpoint_v3 "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+	envoy_extensions_access_loggers_grpc_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/access_loggers/grpc/v3"
+	envoy_upstreams_http_v3 "github.com/envoyproxy/go-control-plane/envoy/extensions/upstreams/http/v3"
+	"github.com/hashicorp/consul/acl"
+	"github.com/hashicorp/consul/api"
+	cmn "github.com/hashicorp/consul/envoyextensions/extensioncommon"
+	"github.com/hashicorp/go-multierror"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+const (
+	LocalAccessLogClusterName = "local_access_log"
+
+	localhost     = "localhost"
+	localhostIPv4 = "127.0.0.1"
+	localhostIPv6 = "::1"
+)
+
+type AccessLog struct {
+	LogName                 string
+	GrpcService             *GrpcService
+	BufferFlushInterval     *time.Duration
+	BufferSizeBytes         uint32
+	FilterStateObjectsToLog []string
+	RetryPolicy             *RetryPolicy
+	Body                    interface{}
+	Attributes              map[string]interface{}
+	ResourceAttributes      map[string]interface{}
+}
+
+func (a *AccessLog) normalize() error {
+	if a.GrpcService != nil {
+		a.GrpcService.normalize()
+	} else {
+		return fmt.Errorf("missing GrpcService")
+	}
+
+	if a.RetryPolicy != nil {
+		a.RetryPolicy.normalize()
+	}
+
+	return nil
+}
+
+func (a *AccessLog) validate() error {
+	if a == nil {
+		return nil
+	}
```

Comment:
Is `nil` a valid configuration? It seems like a programming error to me to call `validate` on a `nil` value.

Reply:
Removed the unwanted check